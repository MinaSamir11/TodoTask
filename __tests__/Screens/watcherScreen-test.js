import React from 'react';
import {render, act} from '@testing-library/react-native';
import WatcherScreen from '../../src/Screens/Watcher';

jest.mock('react-native-background-timer', () => ({
  stopBackgroundTimer: jest.fn(),
}));

describe('test Watcher Screen', () => {
  it('should Renders WATCHER screen and MatchSnapShot', () => {
    const {toJSON} = render(<WatcherScreen />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should Screen has 2 button and textInput', async () => {
    const {getByTestId} = render(<WatcherScreen />);

    await act(async () => {
      expect(getByTestId('WATCHER-SCREEN')).toBeTruthy();
      expect(getByTestId('INPUT-TASK_NAME')).toBeTruthy();
      expect(getByTestId('START-BTN')).toBeTruthy();
      expect(getByTestId('STOP-BTN')).toBeTruthy();
    });
  });
});
