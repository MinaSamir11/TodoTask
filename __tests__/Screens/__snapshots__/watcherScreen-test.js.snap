// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Watcher Screen should Renders WATCHER screen and MatchSnapShot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  testID="WATCHER-SCREEN"
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
        "margin": 15,
      }
    }
  >
    Last time spend in task: null
  </Text>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder="Enter Task Name"
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#F3F3F3",
        "color": "#000",
        "fontWeight": "bold",
        "height": 40,
        "marginTop": 10,
        "paddingHorizontal": 10,
        "width": 600,
      }
    }
    testID="INPUT-TASK_NAME"
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
        "margin": 15,
      }
    }
  >
    Timer:
      
    00:00:00
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "marginBottom": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#18CC04",
            "borderRadius": 8,
            "height": 42,
            "justifyContent": "center",
            "marginEnd": 10,
            "marginStart": 10,
            "width": 225,
          }
        }
        testID="START-BTN"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "letterSpacing": 1,
            }
          }
        >
          Start
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FB3E29",
            "borderRadius": 8,
            "height": 42,
            "justifyContent": "center",
            "marginEnd": 10,
            "marginStart": 10,
            "width": 225,
          }
        }
        testID="STOP-BTN"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "letterSpacing": 1,
            }
          }
        >
          Stop
        </Text>
      </View>
    </View>
  </View>
</View>
`;
